source/extensions/filters/http/rbac/rbac_filter.cc

```diff
@@ -77,24 +77,46 @@ Http::FilterHeadersStatus RoleBasedAccessControlFilter::decodeHeaders(Http::Head
                 callbacks_->connection()->ssl()->subjectPeerCertificate()
           : "none",
       headers, callbacks_->requestInfo().dynamicMetadata().DebugString());
+  std::string effective_policy_id;
   const absl::optional<Filters::Common::RBAC::RoleBasedAccessControlEngineImpl>& shadow_engine =
       config_->engine(callbacks_->route(), EnforcementMode::Shadow);
   if (shadow_engine.has_value()) {
+    std::string shadow_resp_code = "200";
     if (shadow_engine->allowed(*callbacks_->connection(), headers,
-                               callbacks_->requestInfo().dynamicMetadata())) {
+                               callbacks_->requestInfo().dynamicMetadata(), effective_policy_id)) {
       ENVOY_LOG(debug, "shadow allowed");
       config_->stats().shadow_allowed_.inc();
     } else {
       ENVOY_LOG(debug, "shadow denied");
       config_->stats().shadow_denied_.inc();
+      shadow_resp_code = "403";
+    }
+
+    const auto& filter_metadata = callbacks_->requestInfo().dynamicMetadata().filter_metadata();
+    const auto filter_it = filter_metadata.find(HttpFilterNames::get().Rbac);
+    if (filter_it != filter_metadata.end()) {
+      ProtobufWkt::Struct metrics;
+
+      if (effective_policy_id != "") {
```

Comment:
nit: `!effective_policy_id.empty()` (can this happen btw, isn't every policy named even if default?)

Reply:
updated.(when there is no matching policy found in ```allowed``` , effective_policy_id here is default to be empty)